scala2java-core/src/main/scala/io/github/effiban/scala2java/core/predicates/TermSelectHasApplyMethod.scala

```diff
@@ -12,22 +12,12 @@ object TermSelectHasApplyMethod extends TermSelectHasApplyMethod {
   override def apply(termSelect: Term.Select): Boolean = {
     (termSelect.qual, termSelect.name) match {
       case (qual: Term.Ref, name) =>
-        hasApplyMethodOnObjectOf(termSelect) ||
-        hasApplyMethodOnCompanionObjectOfType(Type.Select(qual, Type.Name(name.value)))
+        hasApplyMethodOnObjectOf(termSelect)
       case _ => false
     }
   }
 
   private def hasApplyMethodOnObjectOf(termSelect: Term.Select) = {
     isTermMemberOf(termSelect, q"apply")
   }
-
-  private def hasApplyMethodOnCompanionObjectOfType(typeSelect: Type.Select) = {
-    // This is a "smart hack".
-    // Sometimes the code will define a variable referencing a Scala object (type singleton), and we cannot resolve that by reflection.
-    // However usually the object in question is a companion object, and its associated type also has an alias
-    // with the same qualified name. That alias can be resolved by reflection, so we can use it to reach the desired object.
-    // Example: "scala.List" refers to "scala.collection.immutable.List" for both the class and the object.
-    isTermMemberOfCompanionOf(typeSelect, q"apply")
-  }
```

Comment:
No longer necessary, since we do the dealiasing in a new method that is called only when qualifying